app/src/main/java/woowacourse/shopping/ui/MainActivity.kt

```
@@ -0,0 +1,103 @@
+                        val currentRoute = navController.currentBackStackEntry?.destination?.route
+
+                        if (currentRoute?.contains("Payment", ignoreCase = true) == true) {
+                            navigated = true
+                            break
+                        }
+                        navController.navigate(PaymentRoute(selectedItemIds = pendingSelectedItemIds))
+                        delay(100)
+
+                        val afterRoute = navController.currentBackStackEntry?.destination?.route
+                        if (afterRoute?.contains("Payment", ignoreCase = true) == true) navigated = true
+                    } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)